File: 2015/01/solution.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string line;
    cin >> line;
    int floor = 0, basement = 0, i = 1;
    for (char ch : line) {
        if (ch == '(') {
            floor++;
    	} else {
            floor--;
	}
        if (basement == 0 && floor == -1) {
            basement = i;
        }
        i++;
    }
    cout << "Part One: " << floor << "\n";
    cout << "Part Two: " << basement << "\n";
    return 0;
}

File: 2015/01/solution.py
line = input()
floor, basement = 0, 0
for i, ch in enumerate(line, start=1):
    floor += 1 if ch == '(' else -1
    if basement == 0 and floor == -1:
        basement = i
print("Part One:", floor)
print("Part Two:", basement)
